pkg/controller/vsphere/machine_scope.go

```diff
@@ -78,19 +81,25 @@ func newMachineScope(params machineScopeParams) (*machineScope, error) {
 
 // Patch patches the machine spec and machine status after reconciling.
 func (s *machineScope) PatchMachine() error {
-	klog.V(3).Infof("%v: patching", s.machine.GetName())
-	// TODO: patch machine
+	klog.V(3).Infof("%v: patching machine", s.machine.GetName())
 
+	// patch status
 	providerStatus, err := apivshpere.RawExtensionFromProviderStatus(s.providerStatus)
 	if err != nil {
 		return machineapierros.InvalidMachineConfiguration("failed to get machine provider status: %v", err.Error())
 	}
-
 	s.machine.Status.ProviderStatus = providerStatus
-	if err := s.client.Status().Patch(context.Background(), s.machine, s.machineToBePatched); err != nil {
-		klog.Errorf("Failed to update machine %q: %v", s.machine.GetName(), err)
+	if err := s.client.Status().Patch(context.Background(), s.machine.DeepCopy(), s.machineToBePatched); err != nil {
```

Comment:
Not sure I understand the design of the machineScope. Why do we deep copy the machine here? That means we discard the results of the update, and now both the `machine` and `machineToBePatched` fields are out of date.